pkg/autodiscovery/common/utils/prometheus.go

```diff
@@ -21,14 +22,27 @@ const (
 // buildInstances generates check config instances based on the Prometheus config and the object annotations
 // The second returned value is true if more than one instance is found
 func buildInstances(pc *types.PrometheusCheck, annotations map[string]string, namespacedName string) ([]integration.Data, bool) {
+	openmetricsVersion := config.Datadog.GetInt("prometheus_scrape.version")
+
 	instances := []integration.Data{}
 	for k, v := range pc.AD.KubeAnnotations.Incl {
 		if annotations[k] == v {
 			log.Debugf("'%s' matched the annotation '%s=%s' to schedule an openmetrics check", namespacedName, k, v)
 			for _, instance := range pc.Instances {
 				instanceValues := *instance
-				if instanceValues.URL == "" {
-					instanceValues.URL = types.BuildURL(annotations)
+				if instanceValues.PrometheusURL == "" && instanceValues.OpenMetricsEndpoint == "" {
+					switch openmetricsVersion {
+					case 1:
+						instanceValues.PrometheusURL = types.BuildURL(annotations)
+						if len(instanceValues.Metrics) == 1 && instanceValues.Metrics[0] == "PLACEHOLDER" {
```

Comment:
Would it make sense to make this a method of instanceValues?

Also, I think that comparing slice identity might work better here. It is of course unlikely that anyone would want a metric named "PLACEHOLDER", but why leave it to chance?
```suggestion
						if len(instanceValues.Metrics) == len(DefaultPrometheusCheck.Metrics) && &instanceValues.Metrics[0] == &DefaultPrometheusCheck.Metrics[0] {
```